tests/random_attack.py

```diff
@@ -707,6 +739,23 @@ def add_range_column(self, name, rangeobj):
         self.dedup_names()
         self.df.cbind(dt.Frame(rangeobj, names=[name]))
 
+    def set_key_columns(self, keys, names):
+        try:
+            self.df.key = names
+        except ValueError as e:
+            assert str(e) == "Cannot set a key: the values are not unique"
+            return
+
+        nonkeys = sorted(list(set(range(len(self.data))) - set(keys)))
+
+        self.types = [self.types[i] for i in keys] + [self.types[i] for i in nonkeys]
+        self.names = [self.names[i] for i in keys] + [self.names[i] for i in nonkeys]
+
+        if (len(self.data[0])):
```

Comment:
```suggestion
        if self.nrows > 0:
```

Reply:
Not sure we should use frame data here either.